### PR TITLE
#46: Added new user settings

### PR DESCRIPTION
**Pull request by acreations**
Added a new settings page for users. In this PR it will only be a container without any actions done

image

### CODE REVIEW

File: public/locales/en-US/translation.json [hunk 1]
```diff
@@ -93,7 +93,6 @@
     "forgotPasswordResetButton": "Reset",
     "goTo": "To",
     "loginButton": "Login",
-    "logout": "Log out",
```
Comment (cebed) [inline public/locales/en-US/translation.json:96]: Put it back

File: public/locales/sv-SE/translation.json [hunk 2]
```diff
@@ -93,7 +93,6 @@
     "forgotPasswordResetButton": "Återställ",
     "goTo": "Till",
     "loginButton": "Logga in",
-    "logout": "Logga ut",
```
Comment (cebed) [inline public/locales/sv-SE/translation.json:96]: Put it back in sv-SE also

File: src/components/settings/UserGroups.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import Box from '@mui/material/Box';
```
Comment (cebed) [inline src/components/settings/UserGroups.tsx:1]: you could remove one import


`import { Box, Typography }  from '@mui/material;`

File: src/components/settings/UserRoles.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import Box from '@mui/material/Box';
```
Comment (cebed) [inline src/components/settings/UserRoles.tsx:1]: Same here

import { Box, Typography }  from '@mui/material;

File: src/components/settings/Users.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
+import { CancelOutlined, CheckCircleOutline } from '@mui/icons-material';
```
Comment (cebed) [inline src/components/settings/Users.tsx:1]: Sort the imports in order to maintain clean code

```javascript 
import React from 'react';
import { CancelOutlined, CheckCircleOutline, Delete } from '@mui/icons-material';
import {
    Avatar,
    Chip,
    IconButton,
    Paper,
    Stack,
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableHead,
    TableRow,
    Typography,
} from '@mui/material';
import { User, UserRole, UserTenantPermissionsGroup } from 'tieraid-api';

import { UsersType } from 'services/resources/users';
import { shallow } from 'zustand/shallow';
import { useTranslation } from 'react-i18next';
import { useUserStore } from 'store/userStore';
```

File: src/components/settings/Users.tsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                            <TableCell>{t('users.verifiedLabel')}</TableCell>
+                            <TableCell>{t('users.rolesLabel')}</TableCell>
+                            <TableCell>{t('users.groupsLabel')}</TableCell>
+                            <TableCell></TableCell>
+                        </TableRow>
+                    </TableHead>
+                    <TableBody>
+                        {userList.map((user, index) => (
+                            <TableRow
+                                key={user.email || ''}
```
Comment (cebed) [inline src/components/settings/Users.tsx:82]: `user?.email` instead

safely access properties of the object without causing an error if the object in null or something

File: src/components/settings/Users.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                            <TableCell></TableCell>
+                            <TableCell></TableCell>
+                            <TableCell>{t('users.verifiedLabel')}</TableCell>
+                            <TableCell>{t('users.rolesLabel')}</TableCell>
+                            <TableCell>{t('users.groupsLabel')}</TableCell>
+                            <TableCell></TableCell>
+                        </TableRow>
+                    </TableHead>
+                    <TableBody>
+                        {userList.map((user, index) => (
```
Comment (cebed) [inline src/components/settings/Users.tsx:80]: `userList?.map` instead

safely access without causing an error if the list is null or something

Comment (acreations) [reply src/components/settings/Users.tsx:80]: userList cannot be undefined according to TypeScript

File: public/locales/en-US/translation.json (hunk 1 above)
Comment (acreations) [reply public/locales/en-US/translation.json:96]: We need to add translation directly when created the MENU_OPTIONS

**Comment by cebed** [comment]
Looks Good

**Pull request merged**